src/UKHO.S100PermitService.API/Controllers/PermitController.cs

```diff
@@ -21,6 +34,37 @@ public virtual async Task<IActionResult> GeneratePermits(int licenceId)
         {
             _logger.LogInformation(EventIds.GeneratePermitStarted.ToEventId(), "Generate Permit API call started.");
 
+            var productsList = new List<products>();
+            productsList.Add(new products()
+            {
+                id = "ID",
+                datasetPermit = new productsProductDatasetPermit[]
+                {
+                    new productsProductDatasetPermit() {
+                        issueDate = DateTimeOffset.Now.ToString("yyyy-MM-ddzzz"),
+                        editionNumber = 1,
+                        encryptedKey = "encryptedkey",
+                        expiry = DateTime.Now,
+                        filename = "filename",
+
+                    }
+                }
+            });
+            var upn = "ABCDEFGHIJKLMNOPQRSTUVYXYZ";
+            var tempPath = Path.Combine(Path.GetTempPath(), "Master", $"PERMIT.xml");
+
+            _logger.LogInformation(EventIds.MapDataToPermitStart.ToEventId(), "MapDataToPermit call started");
+            var permit = _permitXmlService.MapDataToPermit(DateTimeOffset.Now, "AB", "ABC", upn, 1.0m, productsList);
+            _logger.LogInformation(EventIds.MapDataToPermitEnd.ToEventId(), "MapDataToPermit call completed");
+
+            _logger.LogInformation(EventIds.XmlSerializationStart.ToEventId(), "Permit Xml serialization started");
+            var permitXml = _xmlHelper.GetPermitXmlString(permit);
+            _logger.LogInformation(EventIds.XmlSerializationEnd.ToEventId(), "Permit Xml serialization completed");
+
+            _logger.LogInformation(EventIds.FileCreationStart.ToEventId(), "Xml file creation started");
+            _fileSystemHelper.CreateFile(permitXml, tempPath);
```

Comment:
See comments about the helper classes.

_permitReaderWriter.WritePermit(permit); 

Knowledge of the tempPath could be encapsulated in that reader/writer object?

Reply:
Added the method